client/src/utils/requests.js

```diff
@@ -0,0 +1,159 @@
+import axios from 'axios';
+import { checkHistory, checkLikes, checkWatchLater } from './utils';
+
+export const handleFetchVideos = async (url) => {
+   const response = await axios({ method: 'GET', url: url });
+
+   if (response.status === 200 || response.status === 201) {
+      return response;
+   } else {
+      throw new Error('Failed to fetch videos');
+   }
+};
+
+export const handleAddRemoveLike = async ({
+   state,
+   dispatch,
+   video,
+   notify,
+}) => {
+   if (checkLikes(state, video).length === 0) {
+      try {
+         const { status } = await axios({
+            method: 'POST',
+            url: `https://driftview-backend.duhbhavesh.repl.co/liked`,
+            data: {
+               _id: video.id,
+            },
+         });
+
+         if (status === 200 || status === 201) {
+            dispatch({ type: 'ADD_TO_LIKED', payload: video });
+         }
+         notify('Added to Liked Videos');
+      } catch (err) {
+         console.log(err);
+      }
+   } else {
+      try {
+         const { status } = await axios({
+            method: 'DELETE',
+            url: `https://driftview-backend.duhbhavesh.repl.co/liked/${video.id}`,
+         });
+
+         if (status === 200 || status === 201) {
+            dispatch({ type: 'REMOVE_FROM_LIKED', payload: video });
+         }
+         notify('Removed from Liked Videos');
+      } catch (err) {
+         console.log(err);
+      }
+   }
+};
+
+export const handleRemoveLike = async ({ dispatch, video }) => {
+   try {
+      const { status } = await axios({
+         method: 'DELETE',
+         url: `https://driftview-backend.duhbhavesh.repl.co/liked/${video.id}`,
+      });
+
+      if (status === 200 || status === 201) {
+         dispatch({ type: 'REMOVE_FROM_LIKED', payload: video });
+      }

```

Comment:
when you are doing the same thing in both these part..then why don't you call this function itself in else

Reply:
Yes, I have that already in else, I need this function to remove like from the listing page.